models/classes/runner/QtiRunnerService.php

```diff
@@ -1132,6 +1132,18 @@ public function skip(RunnerServiceContext $context, $scope, $ref)
         return $this->move($context, 'skip', $scope, $ref);
     }
 
+    /**
+     * @param RunnerServiceContext $context
+     * @param $scope
+     * @param $ref
+     * @return boolean
+     * @throws \common_Exception
+     */
+    public function autoSkip(RunnerServiceContext $context, $scope, $ref)
```

Comment:
Do you mean by skipping to submit a default value response? That's the equivalent in our implementation because the implementation in CG is faulty, but in NG, skipping doesn't mean that you submit a default value response. Maybe another name is better. What we want to do with this is to submit a default value for non-presented items.

Reply:
This will submit a default value indeed. However, I suggest we leave the name as it is as it better fits this particular abstraction layer terminology

Reply:
ok after thinking over that for a while I think you're right :) we should differentiate the faulty implementation from this functionality. Renamed now `autoSkipWithDefaultResponse`